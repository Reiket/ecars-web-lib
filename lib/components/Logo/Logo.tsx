import type {FC} from 'react';
import {cn} from '@/services/helpers';
import type {ElementProps} from '@/services/types';

interface Props extends ElementProps {
  src: string;
}

export const Logo: FC<Props> = ({src, block}) => (
  <a
    href="/"
    className={cn(block, 'logo')}
  >
    <img
      src={src}
      alt="Logo"
    />
  </a>
);
