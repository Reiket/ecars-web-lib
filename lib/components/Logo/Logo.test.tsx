import {Logo} from '@/components/Logo/Logo';
import {render, screen} from '@testing-library/react';

describe('Logo Component', () => {
  test('render components correctly with src attr', () => {
    const mockSrc = 'https://example.com/logo.png';
    const {container} = render(<Logo src={mockSrc} />);
    const img = screen.getByRole('img');
    expect(img).toHaveAttribute('src', mockSrc);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
