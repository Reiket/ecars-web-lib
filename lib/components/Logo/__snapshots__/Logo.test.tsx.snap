// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo Component > render components correctly with src attr 1`] = `
<div>
  <a
    class="logo"
  >
    <img
      alt="Logo"
      src="https://example.com/logo.png"
    />
  </a>
</div>
`;
